src/order/bounds.lean

```diff
@@ -1023,6 +1024,174 @@ hf.dual_left.le_is_glb_image ha hb
 
 end antitone
 
+section image2
+variables [preorder α] [preorder β] [preorder γ] {f : α → β → γ} {s : set α} {t : set β} {a : α}
+  {b : β}
+
+section monotone_monotone
+variables (h₀ : ∀ b, monotone (swap f b)) (h₁ : ∀ a, monotone (f a))
+include h₀ h₁
+
+lemma mem_upper_bounds_image2 (ha : a ∈ upper_bounds s) (hb : b ∈ upper_bounds t) :
+  f a b ∈ upper_bounds (image2 f s t) :=
+forall_image2_iff.2 $ λ x hx y hy, (h₀ _ $ ha hx).trans $ h₁ _ $ hb hy
+
+lemma mem_lower_bounds_image2 (ha : a ∈ lower_bounds s) (hb : b ∈ lower_bounds t) :
+  f a b ∈ lower_bounds (image2 f s t) :=
+forall_image2_iff.2 $ λ x hx y hy, (h₀ _ $ ha hx).trans $ h₁ _ $ hb hy
+
+lemma image2_upper_bounds_upper_bounds_subset :
+  image2 f (upper_bounds s) (upper_bounds t) ⊆ upper_bounds (image2 f s t) :=
+by { rintro _ ⟨a, b, ha, hb, rfl⟩, exact mem_upper_bounds_image2 h₀ h₁ ha hb }
+
+lemma image2_lower_bounds_lower_bounds_subset :
+  image2 f (lower_bounds s) (lower_bounds t) ⊆ lower_bounds (image2 f s t) :=
+by { rintro _ ⟨a, b, ha, hb, rfl⟩, exact mem_lower_bounds_image2 h₀ h₁ ha hb }
+
+lemma bdd_above.image2 : bdd_above s → bdd_above t → bdd_above (image2 f s t) :=
```

Comment:
This naming is a bit weird because there is no `bdd_above.image`. I think a comment on each one of these lemmas that refers to what the `set.image` analog is called would be helpful.